languageserver/integration/resolvers.go

```diff
@@ -66,6 +72,32 @@ func (r *resolvers) addressContractNames(address common.Address) ([]string, erro
 	return names, nil
 }
 
+// accountAccess checks whether the current program location and accessed program location were deployed to the same account.
+//
+// if the contracts were deployed on the same account then it returns true and hence allows the access, false otherwise.
+func (r *resolvers) accountAccess(checker *sema.Checker, memberLocation common.Location) bool {
+	contracts, err := r.client.getState().DeploymentContractsByNetwork(config.DefaultEmulatorNetwork().Name)
```

Comment:
Thank you for pointing out! The state is loaded in-memory and all it does is iterates over the slice to find the correct values which should be in range of ~10 items. I don't think this should be a performance concern and at this point wouldn't add caching complexity. The flow.json also might change as you add things to it and it will reload the values but that will be done in background so it wouldn't really affect this, but caching might cause some dealys in the UI after adding the values in flow.json which might be confusing. What do you think?